server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader9to10.java

```diff
@@ -151,19 +152,9 @@ private static String extractAuthName(ACL acl) {
     }
   }
 
-  private static boolean canWrite(List<ACL> acls) {
-    if (ZooDefs.Ids.OPEN_ACL_UNSAFE.equals(acls)) {
-      return true;
-    }
-    ;
-    for (ACL acl : acls) {
-      String name = extractAuthName(acl);
-      if (("accumulo".equals(name) || "anyone".equals(name))
-          && acl.getPerms() >= ZooDefs.Perms.WRITE) {
-        return true;
-      }
-    }
-    return false;
+  private static boolean canWrite(final Set<String> users, final List<ACL> acls) {
+    return ZooDefs.Ids.OPEN_ACL_UNSAFE.equals(acls) || acls.stream()
+        .anyMatch(a -> users.contains(extractAuthName(a)) && a.getPerms() >= ZooDefs.Perms.WRITE);
```

Comment:
The perms are bits - so `>= ZooDefs.Perms.WRITE` will return true if any combination of write, create, delete, admin is set - so if the perm was delete only, this would still return "true" for can write, which seems misleading at best.

I think the test might be better names hasAll and the condition was:

```
 .anyMatch(a -> users.contains(extractAuthName(a)) && a.getPerms() == ZooDefs.Perms.ALL;
```